mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -83,6 +86,15 @@ public EntityDefinition getDefinition(TypeElement processedClass) {
       typeHierarchy.add((TypeElement) context.getTypeUtils().asElement(type));
     }
 
+    boolean isScalaCaseClass = isScalaCaseClass(typeHierarchy);
+
+    Optional<IntrospectionStrategy> introspectionStrategy = getIntrospectionStrategy(typeHierarchy);
```

Comment:
From this logic, it seems that when the class is Scala case class, the `GetterStyle` should be `SHORT` and mutable should be `false`.  Shouldn't we validate and forbid the usage of those two parameters IF the `isScalaCaseClass` is true? This will be convention over configuration. 
On the other hand, it is possible that the scala case class has a custom getter (but for sure very uncommon) so we may want to keep that.
Regarding `mutable` I think that `case class` cannot be mutable to this property could be dropped if `isScalaCaseClass == true`

Reply:
The processor sets `getterStyle = SHORT, mutable = false` by default for Scala classes. But if an explicit `@IntrospectionStrategy` annotation is present, it will override those defaults. See how the two variables below are initialized:
```java
    boolean mutable =
        introspectionStrategy.map(IntrospectionStrategy::mutable).orElse(!isScalaCaseClass);
```
This way we default to the most likely behavior for Scala, but you still have the flexibility to override whatever you want.